File: php/class-access/sub/Processor.php
<?php

namespace sub;

require_once(__DIR__ . '/Value.php');

// NOTE: どこからでも利用可能
class Processor
{
    // NOTE: Facade から利用するためには public にするしかない
    //       その結果、他のどのクラスからでも利用可能になってしまう
    public function process(Value $value): void
    {
        echo ($value->value + 1) . PHP_EOL;
    }
}

File: java/class-access/sub/Processor.java
package sub;

// NOTE: パッケージ外から利用不可
class Processor {
    // NOTE: 同一パッケージの Facade からは利用可能
    //       パッケージの外からは利用不可
    void process(Value value) {
        System.out.println(value.value + 1);
    }
}
